### PR TITLE
aot suggested fixes

### PR DESCRIPTION
**Pull request by andreaTP**
At Devoxx I had the pleasure to meet @forax , he is one of the maintainer of the ASM library we are using in the AOT compiler and suggested those fixes.

### CODE REVIEW

File: aot/src/main/java/com/dylibso/chicory/aot/AotMachine.java [hunk 1]
```diff
@@ -367,13 +367,13 @@ private byte[] compileClass(String className, FunctionSection functions) {
         var classWriter = new ClassWriter(ClassWriter.COMPUTE_FRAMES);
         classWriter.visit(
                 Opcodes.V11,
-                Opcodes.ACC_PUBLIC | Opcodes.ACC_FINAL,
+                Opcodes.ACC_PUBLIC | Opcodes.ACC_FINAL | Opcodes.ACC_SUPER,
```
Comment (andreaTP) [inline aot/src/main/java/com/dylibso/chicory/aot/AotMachine.java:370]: Apparently, compatibility with project Leyden.

File: aot/src/main/java/com/dylibso/chicory/aot/AotMachine.java [hunk 2]
```diff
@@ -367,13 +367,13 @@ private byte[] compileClass(String className, FunctionSection functions) {
...
                 internalClassName,
                 null,
                 getInternalName(Object.class),
                 new String[] {getInternalName(Machine.class)});
 
-        classWriter.visitSource("wasm", "wasm");
+        classWriter.visitSource(null, null);
```
Comment (andreaTP) [inline aot/src/main/java/com/dylibso/chicory/aot/AotMachine.java:376]: Better to leave those fields empty than using dummy values.

File: aot/src/main/java/com/dylibso/chicory/aot/AotMachine.java (hunk 1 above)
Comment (forax) [reply aot/src/main/java/com/dylibso/chicory/aot/AotMachine.java:370]: Not Leyden, Valhalla. In Valhalla, if you have a class without ACC_SUPER (renamed ACC_IDENTITY), you are creating a value class. See https://bugs.openjdk.org/browse/JDK-8317278

Comment (andreaTP) [reply aot/src/main/java/com/dylibso/chicory/aot/AotMachine.java:370]: Many thanks for the explanation!

Comment (electrum) [reply aot/src/main/java/com/dylibso/chicory/aot/AotMachine.java:370]: I'm guessing this doesn't matter was we're currently generating `V11` classes, but good to fix for the future!

**Review by electrum — APPROVED** [review]

File: aot/src/main/java/com/dylibso/chicory/aot/AotMachine.java (hunk 2 above)
Comment (electrum) [reply aot/src/main/java/com/dylibso/chicory/aot/AotMachine.java:376]: The second argument seems wrong for sure. The first one causes stack traces to include `wasm`, which seems useful, but I don't know if there are downsides. If @forax says we should remove it, then let's remove it!

Comment (electrum) [inline aot/src/main/java/com/dylibso/chicory/aot/AotMachine.java:376]: We can simply remove this call, as calling with null won't do anything.

Comment (forax) [reply aot/src/main/java/com/dylibso/chicory/aot/AotMachine.java:376]: As electrum said, for me, it should be
```java
classWriter.visitSource("wasm", null);
```
so you get "wasm" printed in the stacktraces.

Comment (andreaTP) [reply aot/src/main/java/com/dylibso/chicory/aot/AotMachine.java]: Ok, I have nothing against, my reasoning was:
"better to not provide anything than a wrong `fileName` that the IDE will try to resolve".

But the IDE is still broken in either cases, so I just opened #576 to track it as a separate issue.

Thanks everyone for the feedback, much appreciated!

**Pull request merged**